app/src/main/java/com/concordium/wallet/ui/cis2/retrofit/MetadataApi.kt

```diff
@@ -4,7 +4,11 @@ import com.concordium.wallet.data.model.TokenMetadata
 import retrofit2.Response
 import retrofit2.http.GET
 import retrofit2.http.Url
+import retrofit2.http.Query
 
 interface MetadataApi {
-    @GET suspend fun metadata(@Url url: String?): Response<TokenMetadata>
+    /**
+     * @param metadataChecksum is a special parameter only used by the interceptor to ensure the response body has that checksum
+     */
+    @GET suspend fun metadata(@Url url: String?, @Query("metadataChecksum") metadataChecksum: String?): Response<TokenMetadata>
```

Comment:
This might be too hacky, but I struggled to find a nice way to verify the checksum against the raw response body.